src/main/java/com/treasuredata/client/impl/ProxyAuthenticator.java

```diff
@@ -20,44 +20,50 @@
 
 import com.google.common.base.Optional;
 import com.treasuredata.client.ProxyConfig;
-import org.eclipse.jetty.client.api.Authentication;
-import org.eclipse.jetty.client.api.Request;
-import org.eclipse.jetty.http.HttpHeader;
-import org.eclipse.jetty.util.B64Code;
+import com.treasuredata.client.TDClientException;
+import com.treasuredata.client.TDClientHttpException;
+import okhttp3.Authenticator;
+import okhttp3.Credentials;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.Route;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.net.URI;
-import java.nio.charset.StandardCharsets;
+import java.io.IOException;
 
 /**
  *
  */
-public class ProxyAuthResult
-        implements Authentication.Result
+public class ProxyAuthenticator
+        implements Authenticator
 {
-    private final Logger logger = LoggerFactory.getLogger(ProxyAuthResult.class);
+    private final Logger logger = LoggerFactory.getLogger(ProxyAuthenticator.class);
     private final ProxyConfig proxyConfig;
     private Optional<String> proxyAuthCache = Optional.absent();
 
-    public ProxyAuthResult(ProxyConfig proxyConfig)
+    public ProxyAuthenticator(ProxyConfig proxyConfig)
     {
         this.proxyConfig = proxyConfig;
     }
 
     @Override
-    public URI getURI()
+    public Request authenticate(Route route, Response response)
+            throws IOException
     {
-        return proxyConfig.getUri();
-    }
+        if (response.request().header("Proxy-Authorization") != null) {
```

Comment:
Without this line, OkHttp enters infinite-loop if a wrong password is used for 
 proxy servers: https://github.com/square/okhttp/wiki/Recipes#handling-authentication